lls/main.c

```diff
@@ -16,11 +16,209 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "gatekeeper_lls.h"
+#include <rte_cycles.h>
+#include <rte_ethdev.h>
+
+#include "cache.h"
+
+/* Length of time (in seconds) to wait between scans of the cache. */
+#define LLS_CACHE_SCAN_INTERVAL 10
+
+static struct lls_config lls_conf;
+
+struct lls_config *
+get_lls_conf(void)
+{
+	return &lls_conf;
+}
+
+static int
+cleanup_lls(void)
+{
+	/* TODO Destroy ARP and ND cache. */
+	destroy_mailbox(&lls_conf.requests);
+	rte_timer_stop(&lls_conf.timer);
+	return 0;
+}
+
+static void
+lls_scan(__attribute__((unused)) struct rte_timer *timer,
+	__attribute__((unused)) void *arg)
+{
+	/* TODO Scan the ARP and ND caches. */
```

Comment:
I see the problem here. The ARP cache hasn't been defined yet.

In this case, we have another problem: we're potentially breaking the binary search of git because if `lls_cache_scan()` has a bug, only the next patch will trigger it.